### ISSUE TITLE
Contract verification on Optimism Goerli displays wrong URL

### ISSUE THREAD

**Issue by WarTech9** [NomicFoundation/hardhat#3288]
After verifying a contract with hardhat-etherscan plugin, a message is displayed showing the successful status and the URL to the newly verified contract on Etherscan:

```
Successfully verified contract ${contractName} on Etherscan.
https://goerli-optimistic.etherscan.io/address/${contractAddress}#code
```

The URL displayed points to https://goerli-optimistic.etherscan.io

Opening this URL does not work because the block explorer URL for goerli is https://goerli-optimism.etherscan.io/ (optimism not optimistic). 

While Etherscan seems to have a redirect from optimistic to optimism, this does not include the path after redirecting, thus, opening the link displayed does not take you to the page for your contract.

**Comment by fvictorio** [comment]
Great catch, thanks a lot for reporting this @WarTech9!

**Comment by fvictorio** [comment]
Fixed in [@nomiclabs/hardhat-etherscan@3.1.2](https://github.com/NomicFoundation/hardhat/releases/tag/%40nomiclabs%2Fhardhat-etherscan%403.1.2).
